consensus/service/src/byzantine_ledger.rs

```diff
@@ -76,16 +75,14 @@ pub struct ByzantineLedger {
 
 impl ByzantineLedger {
     pub fn new<
-        E: ConsensusEnclaveProxy,
         PC: BlockchainConnection + ConsensusConnection + 'static,
         L: Ledger + Sync + 'static,
-        UI: UntrustedInterfaces + Send + Sync + 'static,
     >(
         node_id: NodeID,
         quorum_set: QuorumSet,
         peer_manager: ConnectionManager<PC>,
         ledger: L,
-        tx_manager: TxManager<E, L, UI>,
+        tx_manager: Arc<Mutex<Box<dyn TxManager>>>,
```

Comment:
Previously, TxManager was basically a singleton "service" that maintained its internal state behind Arc<Mutex<_>>. That tightly coupled things like ByzantineLedger to TxManager's internal implementation. For example, ByzantineLedger cared about TxManager's type parameters E and UI. ByzantineLedger could not be used or tested independently of the particular implementation of TxManager.

Now, TxManager is a trait, which decouples ByzantineLedger from its implementation. The Arc<Mutex<_>> has been brought up to this level, which is consistent with the idea that TxManager is a "singleton service", similar to other things like `broadcaster`.